specification/resources/resource-manager/Microsoft.Resources/preview/2022-11-01-preview/snapshots.json

```diff
@@ -0,0 +1,247 @@
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}/providers/Microsoft.Resources/snapshots/{snapshotId}": {
+      "get": {
+        "tags": [
+          "GetSnapshot"
+        ],
+        "operationId": "Snapshots_Get",
+        "x-ms-examples": {
+          "GetSnapshot": {
+            "$ref": "./examples/GetSnapshot.json"
+          }
+        },
+        "description": "Obtains the specified snapshot for the target resource",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceProviderNamespaceParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceTypeParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/snapshotIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/Snapshot"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Snapshot": {
+      "type": "object",
+      "description": "A generic resource snapshot with additional metadata",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SnapshotProperties"
+        }
+      }
+    },
+    "SnapshotProperties": {
+      "description": "The properties of a snapshot",
+      "type": "object",
+      "properties": {
+        "apiVersion": {
+          "type": "string",
+          "description": "The API Version of the resource captured"
+        },
+        "content": {
+          "description": "The actual ARM Resource",
+          "type": "object"
```

Comment:
I thin you also want `additionalProperties: true` on here, so that it is modeling as a 'free form object':

[https://swagger.io/docs/specification/data-models/dictionaries/#:~:text=OpenAPI%20lets%20you%20define%20dictionaries%20where%20the%20keys,pairs.%20For%20example%2C%20a%20string-to-string%20dictionary%20like%20this%3A](https://swagger.io/docs/specification/data-models/dictionaries/#:~:text=OpenAPI%20lets%20you%20define%20dictionaries%20where%20the%20keys,pairs.%20For%20example%2C%20a%20string-to-string%20dictionary%20like%20this%3A)